inference-engine/src/transformations/include/transformations/rt_info/mask_attribute.hpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief Defines primitives priority attribute
+ * @file primitives_priority_attribute.hpp
+ */
+
+#pragma once
+
+#include <assert.h>
+#include <functional>
+#include <memory>
+#include <string>
+#include <set>
+
+#include <ngraph/node.hpp>
+#include <ngraph/variant.hpp>
+#include <transformations_visibility.hpp>
+
+namespace ngraph {
+
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief each element in vector represents dimension and each element
+ * in set is an id of dimensions which contains zeros.
+ */
+class TRANSFORMATIONS_API Mask : public std::vector<std::set<uint64_t>>,
```

Comment:
It is impossible to reuse indices represented as a set for Gather optimization because the order is also important for Gather, but here we don't maintain it.

Reply:
The base class can be changed in any time to any container. So I would leave base structure change to the future when we  decide to implement gather optimization.